modules/tensor_mechanics/src/materials/CompositeEigenstrain.C

```diff
@@ -0,0 +1,29 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "CompositeEigenstrain.h"
+
+template<>
+InputParameters validParams<CompositeEigenstrain>()
+{
+  InputParameters params = CompositeTensorBase<RankTwoTensor, ComputeStressFreeStrainBase>::validParams();
+  params.addClassDescription("Assemble an Eigenstrain tensor from multiple tensor contributions weighted by material properties");
+  return params;
+}
+
+CompositeEigenstrain::CompositeEigenstrain(const InputParameters & parameters) :
+    CompositeTensorBase<RankTwoTensor, ComputeStressFreeStrainBase>(parameters)
+{
+  initializeDerivativeProperties(_base_name + "elastic_strain");
```

Comment:
Just wanted to make sure, did we settle on `elastic_strain` as the standard name? I seem to remember in some cases it was changed to `mechanical_strain` but I can't remember if it was changed back.